models/user.js

```diff
@@ -4,8 +4,8 @@ var	passportLocal=require("passport-local-mongoose");
 var userSchema=new mongoose.Schema({
 
 	//TODO: DEFINE USERNAME AND PASSSWORD ATTRIBUTES
-
-
+    username:String,
+    password:String,
     loaned_books:[
         //TODO: embed reference to id's of book copies loaned by this particular user in this array
```

Comment:
Reference to id of book copies loaned by a user not embedded.

Reply:
Actually I didn't know that reference thing back then so I used an array in the schema and then inserted the book id using this code :)
`await User.findByIdAndUpdate(userId,{$pull:{loaned_books:bid}})`